File: dataStructure/insertionSort/review.py
def insertion_sort(list_of_numbers: list) -> list:
    for i in range(1, len(list_of_numbers)):
        temp = list_of_numbers[i]
        j = i - 1
        while j >= 0 and temp < list_of_numbers[j]:
            list_of_numbers[j + 1], list_of_numbers[ j ] = list_of_numbers[ j ], temp
            j -=1
    return list_of_numbers

print(insertion_sort([9,5,6,4,3,-1]))

File: dataStructure/insertionSort/review.js
const insertionSort = (arrayOfNumbers) => {
  for (let index = 1; index < arrayOfNumbers.length; index++) {
    let temp = arrayOfNumbers[index];
    let j = index - 1;
    while (j >= 0 && temp < arrayOfNumbers[j]) {
      arrayOfNumbers[j + 1] = arrayOfNumbers[j];
      arrayOfNumbers[j] = temp;
      j -= 1;
    }
  }
  return arrayOfNumbers;
};

console.log(insertionSort([-1,-50,50,10,100]));
